mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -7017,17 +7127,118 @@ struct ThreadwiseCopyV2RewritePattern
       SmallVector<Value, 8> destLowerIndices =
           layeredDestIndices[layeredDestIndices.size() - 1];
 
+      bool toEmitOOBStoreCheckLogic = false;
+      SmallVector<unsigned, 2> oobStoreCheckDims;
+      if (composedDestTransform && boundCheckDestAttr) {
+        if (boundCheckDestAttr.size() ==
+            composedDestTransform.getNumResults()) {
+          for (unsigned iter = 0; iter < boundCheckDestAttr.size(); ++iter) {
+            if (boundCheckDestAttr[iter]
+                    .template cast<IntegerAttr>()
+                    .getInt()) {
+              toEmitOOBStoreCheckLogic = true;
+              oobStoreCheckDims.push_back(iter);
+            }
+          }
+        }
+      }
+
       // Store to dest.
       // Issue scalar store.
-      if (dataType == b.getF32Type()) {
-        b.create<StoreOp>(loc, scalarValue, op.dest(), destLowerIndices);
-      } else if (dataType == b.getF16Type()) {
-        auto truncValue = b.create<FPTruncOp>(loc, scalarValue, dataType);
-        b.create<StoreOp>(loc, truncValue, op.dest(), destLowerIndices);
-      } else if (dataType == b.getIntegerType(16)) {
-        auto convertValue =
-            b.create<miopen::DataConvertOp>(loc, dataType, scalarValue);
-        b.create<StoreOp>(loc, convertValue, op.dest(), destLowerIndices);
+      // oob store only support f32 now
+      if (toEmitOOBStoreCheckLogic &&
+          destType.getElementType() == b.getF32Type()) {
+        auto zeroConstantOp = b.create<ConstantIndexOp>(loc, 0);
+        SmallVector<Value, 8> destLowerStoreIndices;
+        SmallVector<Value, 8> destLowerStoreOOBIndices;
+        for (int i = 0; i < destLowerIndices.size(); i++) {
```

Comment:
nit picking: use `unsigned` or `size_t` to remove compile time warnings. also it's a good practice in C++ to use `++i` instead of `i++`.